storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogMetadataManager.java

```diff
@@ -242,4 +242,18 @@ default Optional<RemoteLogSegmentMetadata> nextSegmentWithTxnIndex(TopicIdPartit
     default boolean isReady(TopicIdPartition topicIdPartition) {
         return true;
     }
+
+    /**
+     * Register a listener for when the broker is ready to serve requests.
+     * This method will be called during initialization if the implementation 
+     * supports delayed initialization until the broker can process requests.
+     * 
+     * The default implementation does nothing, allowing implementations that don't 
+     * need delayed initialization to ignore this method.
+     * 
+     * @param brokerReadyFuture A CompletableFuture that completes when broker is ready to serve requests
+     */
+    default void onBrokerReadyForRequests(CompletableFuture<Void> brokerReadyFuture) {
```

Comment:
RemoteLogMetadataManager is a public API change and require a KIP to add a new method.

Reply:
Ok. Got it. Thanks

Reply:
KIP created.